lib/miq_expression/field.rb

```diff
@@ -82,6 +82,19 @@ def contains(other)
     )
   end
 
+  def custom_attribute_column_name
+    column.gsub(CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX, "")
+  end
+
+  def column_type
+    if custom_attribute_column?
+      field_type = CustomAttribute.find_by(:name => custom_attribute_column_name).try(:serialized_value).class
+      field_type == NilClass ? :string : field_type.to_s.downcase.to_sym
```

Comment:
@imtayadeway 
* Removed the '=='.
* Added a method to the custom_attribute class that will determine the field_type.
* Added tests for that method.
* currently I'm not enforcing any type checks, but if they wont fit any of the known types (https://github.com/ManageIQ/manageiq/blob/master/lib/miq_expression.rb#L1359) it would just fall back to string, would you rather have a white list?